core/frontend/src/components/kraken/ExtensionCreationDialog.vue

```diff
@@ -0,0 +1,176 @@
+<template>
+  <v-dialog
+    v-if="new_extension"
+    width="500"
+    :value="Boolean(extension)"
+    persistent
+    @input="showDialog"
+    @click:outside="closeDialog"
+  >
+    <v-card>
+      <v-card-title>
+        Create/Change Extension
+      </v-card-title>
+
+      <v-card-text class="d-flex flex-column">
+        <v-form
+          ref="form"
+          lazy-validation
+        >
+          <v-text-field
+            v-model="new_extension.identifier"
+            label="Extemnsion Identifier"
+            :rules="[validate_identifier]"
+          />
+
+          <v-text-field
+            v-model="new_extension.name"
+            label="Extension Name"
+            :rules="[validate_name]"
+          />
+
+          <v-text-field
+            v-model="new_extension.docker"
+            label="Docker image"
+            :rules="[validate_dockerhub]"
+          />
+
+          <v-text-field
+            v-model="new_extension.tag"
+            label="Docker tag"
+            :rules="[validate_tag]"
+          />
+
+          <v-textarea
+            v-if="new_extension.permissions"
+            v-model="new_extension.permissions"
+            label="Original Permissions"
+            :readonly="true"
+            :rules="[validate_permissions]"
+          />
+
+          <v-textarea
+            v-model="new_extension.user_permissions"
+            label="Custom Permissions (These replace regular permissions)"
+            :rules="[validate_permissions]"
+          />
+
+          <v-btn
+            color="primary"
+            class="mr-4"
+            @click="saveExtension"
+          >
+            Create
+          </v-btn>
+        </v-form>
+      </v-card-text>
+    </v-card>
+  </v-dialog>
+</template>
+
+<script lang="ts">
+import Vue, { PropType } from 'vue'
+
+import Notifier from '@/libs/notifier'
+import { kraken_service } from '@/types/frontend_services'
+import { VForm } from '@/types/vuetify'
+import back_axios from '@/utils/api'
+
+import { InstalledExtensionData } from '../../types/kraken'
+
+const notifier = new Notifier(kraken_service)
+
+export default Vue.extend({
+  name: 'ExtensionCreationDialog',
+  model: {
+    prop: 'extension',
+    event: 'change',
+  },
+  props: {
+    extension: {
+      type: Object as PropType<InstalledExtensionData | null>,
+      default: null,
+    },
+  },
+
+  data() {
+    return {
+      new_extension: this.extension,
+    }
+  },
+  computed: {
+    form(): VForm {
+      return this.$refs.form as VForm
+    },
+  },
+  watch: {
+    new_extension() {
+      this.$emit('input', this.new_extension)
+    },
+    extension() {
+      this.new_extension = this.extension
+    },
+  },
+  methods: {
+    closeDialog() {
+      this.new_extension = null
+    },
+    validate_identifier(input: string): (true | string) {
+      const re = /[A-Za-z]+[A-Za-z0-9]*\.[A-Za-z]+[A-Za-z0-9]*/
```

Comment:
`[A-Za-z]+[A-Za-z0-9]*` is weird, why not `[A-Za-z][A-Za-z0-9]*` ? The plus characters does not make sense to me.
Can you add an example if what this is validating ?

Reply:
yeah, in hindsight the + is meaningless. it is there so that the identifiers do not start with numbers